assembly-syntax/src/library/mod.rs

```diff
@@ -138,6 +138,13 @@ impl Library {
     pub fn mast_forest(&self) -> &Arc<MastForest> {
         &self.mast_forest
     }
+
+    /// Returns the digest of the procedure with the specified name, or `None` if it was not found
+    /// in the Library.
+    pub fn get_procedure_root_by_name(&self, proc_name: QualifiedProcedureName) -> Option<Word> {
+        let node_id = self.exports.get(&proc_name);
+        node_id.map(|id| self.mast_forest()[*id].digest())
+    }
```

Comment:
Should we use something like `TryInto<QualifiedProcedureName>` for `proc_name` parameter? Then, if we have conversion from strings for `QualifiedProcedureName`, we could do something like:

```rust
let root = mylib.get_procedure_root_by_name("foo::bar::baz").unwrap();
```

Reply:
@bitwalker also suggested a slightly different approach - something like:
```rust
pub fn get_procedure_root_by_name(&self, proc_name: AsRef<str>) -> Option<Word>
```
Though, I think `TryInto<QualifiedProcedureName>` covers all the same options, and maybe is a bit more flexible (because we can pass in `QualifiedProcedureName` directly)?

Reply:
I think that it will be useful to be able to pass the `QualifiedProcedureName` directly, because in the `miden-base` I will do exactly that. 

We can't use the string procedure name without the library path (the proc name like `foo` will return an error), so in case we want to get the proc without it (or with the default library path, to be precise) we can just create a `QualifiedProcedureName` directly using `Default::default()` as a library path.

Probably we can use `#anon` path to do the same, but IMO using `QualifiedProcedureName` will be more clean here.